File: framework/math/bounds/Circle.java
package framework.math.bounds;

import framework.math.Vector2;

public class Circle extends Bounds {

    public final Vector2 center;
    public float radius;


    public Circle(float x, float y, float radius) {
        super();
        this.center = new Vector2(x, y);
        this.radius = radius;
    }
}

File: C++/GameEngineCpp/src/physics/Circle.cpp
#include "include/physics/2d-colliders/Circle.hpp"

#include <math/MathUtils.hpp>

using namespace GameEngine::Physics::Bounds;


Circle::Circle(const Vec2& center, double radius)
    : center(center), radius(radius) {}

bool Circle::operator!=(const Circle& other) const
{
    return center != other.center || !isZero(radius - other.radius);
}

bool Circle::operator==(const Circle& other) const
{
    return center == other.center && isZero(radius - other.radius);
}

bool Circle::contains(const Vec2& point) const
{
    return center.distSq(point) <= radius * radius;
}
